RecoLocalTracker/SiPixelRecHits/src/PixelCPEClusterRepair.cc

```diff
@@ -0,0 +1,628 @@
+   
+   //cout << "CPETemplate : " << endl;
+   
+   //--- Default is the maximum error used for edge clusters.
+   //--- (never used, in fact: let comment it out, shut up the complains of the static analyzer, and save a few CPU cycles)
+   //   const float sig12 = 1./sqrt(12.0);
+   //   float xerr = theDetParam.thePitchX *sig12;
+   //   float yerr = theDetParam.thePitchY *sig12;
+   float xerr = 0.0f, yerr = 0.0f;
+   
+   // Check if the errors were already set at the clusters splitting level
+   if ( theClusterParam.theCluster->getSplitClusterErrorX() > 0.0f && theClusterParam.theCluster->getSplitClusterErrorX() < 7777.7f &&
+       theClusterParam.theCluster->getSplitClusterErrorY() > 0.0f && theClusterParam.theCluster->getSplitClusterErrorY() < 7777.7f )
+   {
+      xerr = theClusterParam.theCluster->getSplitClusterErrorX() * micronsToCm;
+      yerr = theClusterParam.theCluster->getSplitClusterErrorY() * micronsToCm;
+      
+      //cout << "Errors set at cluster splitting level : " << endl;
+      //cout << "xerr = " << xerr << endl;
+      //cout << "yerr = " << yerr << endl;
+   }
+   else
+   {
+      // &&& Duplicate from above! Argh!
+      int maxPixelCol = theClusterParam.theCluster->maxPixelCol();
+      int maxPixelRow = theClusterParam.theCluster->maxPixelRow();
+      int minPixelCol = theClusterParam.theCluster->minPixelCol();
+      int minPixelRow = theClusterParam.theCluster->minPixelRow();
+      
+      //--- Are we near either of the edges?
+      bool edgex = ( theDetParam.theRecTopol->isItEdgePixelInX( minPixelRow ) || theDetParam.theRecTopol->isItEdgePixelInX( maxPixelRow ) );
+      bool edgey = ( theDetParam.theRecTopol->isItEdgePixelInY( minPixelCol ) || theDetParam.theRecTopol->isItEdgePixelInY( maxPixelCol ) );
+      
+
+      //--- Check status of both template calls.
+      if ( (theClusterParam.ierr !=0) || (theClusterParam.ierr2 !=0) )   
+      {
+         // If reconstruction fails the hit position is calculated from cluster center of gravity
+         // corrected in x by average Lorentz drift. Assign huge errors.
+         //xerr = 10.0 * (float)theClusterParam.theCluster->sizeX() * xerr;
+         //yerr = 10.0 * (float)theClusterParam.theCluster->sizeX() * yerr;
+         
+         if(!GeomDetEnumerators::isTrackerPixel(theDetParam.thePart))
+            throw cms::Exception("PixelCPEClusterRepair::localPosition :")
+            << "A non-pixel detector type in here?";
+         
+         // Assign better errors based on the residuals for failed template cases
+         if ( GeomDetEnumerators::isBarrel(theDetParam.thePart) )
+         {
+            xerr = 55.0f * micronsToCm;
+            yerr = 36.0f * micronsToCm;
+         }
+         else
+         {
+            xerr = 42.0f * micronsToCm;
+            yerr = 39.0f * micronsToCm;
+         }
+         
+         //cout << "xerr = " << xerr << endl;
```

Comment:
Please remove the commented out lines, or replace with some LogDebug or similar

Reply:
OK, if you require this change now.